### PR TITLE
LPS-127467 Cannot link page in the parent site from a subsite using the item selector

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

I added the ability for the layout selector to display the group selector breadcrumb so that the listed pages will belong to the selected site. I modeled the behavior from [AssetBrowserDisplayContext](https://github.com/liferay/liferay-portal/blob/master/modules/apps/asset/asset-browser-web/src/main/java/com/liferay/asset/browser/web/internal/display/context/AssetBrowserDisplayContext.java).

@robertoDiaz suggested passing the information in `LayoutItemSelectorCriterion` whether the site selector should be displayed, but I wasn't able to find a pattern for that yet.
In the current solution, `isShowBreadcrumb()` checks for the "showBreadcrumb" parameter in the request, and, if the layout selector is displayed in the item selector portlet. If either of these conditions is met (similar to `AssetBrowserDisplayContext`) the breadcrumbs will be shown.

Thank you and best regards,
István

### CODE REVIEW

**Comment by IstvanD** [comment]
ci:test:sf

**Comment by IstvanD** [comment]
ci:test:relevant

**Comment by IstvanD** [comment]
Dear Team,

I was wondering if you could find the time to review the pull request.
Please let me know if you need any help or explanation.

Thank you and best regards,
István

File: modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java [hunk 1]
```diff
@@ -54,12 +90,70 @@ public boolean isPrivateLayout() {
...
+		}
+
+		return false;
+	}
+
 	public boolean isShowHiddenPages() {
 		return _layoutItemSelectorCriterion.isShowHiddenPages();
 	}
 
+	private BreadcrumbEntry _getHomeBreadcrumb() throws PortalException {
```
Comment (adolfopa) [inline modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java:118]: Can you rename this to _getHomeBreadcrumbEntry() ?

File: modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java [hunk 2]
```diff
@@ -54,12 +90,70 @@ public boolean isPrivateLayout() {
...
+				WebKeys.THEME_DISPLAY);
+
+		BreadcrumbEntry breadcrumbEntry = new BreadcrumbEntry();
+
+		breadcrumbEntry.setTitle(themeDisplay.getSiteGroupName());
+
+		return breadcrumbEntry;
+	}
+
+	private BreadcrumbEntry _getSitesAndLibrariesBreadcrumb()
```
Comment (adolfopa) [inline modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java:130]: Same here (_getSitesAndLibrariesBreadcrumbEntry() )

**Review by adolfopa — CHANGES_REQUESTED** [review]
@IstvanD See my comments about minor SF changes. Other than that, this looks good :+1:

File: modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java (hunk 2 above)
Comment (adolfopa) [reply modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java:130]: also, either include the full "Asset Libraries" name in the method (_getSitesAnd*Asset*LibrariesBreadcrumbEntry()), or use the alternative "Depot" (_getSitesAndDepotsBreadcrumbEntry()).

Also, if it returns a single breadcrumb entry, shouldn't the name be singular?

Comment (IstvanD) [reply modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java]: hi @adolfopa I have renamed the methods. The plural in the method name stands for how that single entry is named: "Sites and Libraries" [is displayed](https://github.com/liferay-lima/liferay-portal/blob/f57668f6561cd267ed2a0fd2c8ccaec6122e30ee/modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java#L137) on the screen.

Comment (adolfopa) [reply modules/apps/layout/layout-item-selector-web/src/main/java/com/liferay/layout/item/selector/web/internal/display/context/LayoutItemSelectorViewDisplayContext.java]: Makes sense. Thanks!

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**